x/epoching/keeper/epoch_msg_queue.go

```diff
@@ -107,139 +108,150 @@ func (k Keeper) GetCurrentEpochMsgs(ctx context.Context) []*types.QueuedMessage
 }
 
 // HandleQueuedMsg unwraps a QueuedMessage and forwards it to the staking module
-func (k Keeper) HandleQueuedMsg(ctx context.Context, msg *types.QueuedMessage) (*sdk.Result, error) {
-	var (
-		unwrappedMsgWithType sdk.Msg
-		err                  error
-	)
-	unwrappedMsgWithType = msg.UnwrapToSdkMsg()
-
-	// failed to decode validator address
-	if err != nil {
-		panic(err)
-	}
+func (k Keeper) HandleQueuedMsg(goCtx context.Context, qMsg *types.QueuedMessage) (*sdk.Result, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
 
-	// get the handler function from router
-	handler := k.router.Handler(unwrappedMsgWithType)
+	res, err := k.runUnwrappedMsg(ctx, qMsg)
 
-	// Create a new Context based off of the existing Context with a MultiStore branch
-	// in case message processing fails. At this point, the MultiStore is a branch of a branch.
-	handlerCtx, msCache := cacheTxContext(sdk.UnwrapSDKContext(ctx), msg.TxId, msg.MsgId, msg.BlockHeight)
-
-	// handle the unwrapped message
-	result, err := handler(handlerCtx, unwrappedMsgWithType)
-	if err != nil {
-		return result, err
-	}
+	return sdk.WrapServiceResult(ctx, res, err)
+}
 
-	// release the cache
-	msCache.Write()
+func (k Keeper) runUnwrappedMsg(ctx sdk.Context, qMsg *types.QueuedMessage) (proto.Message, error) {
+	msgSvrCtx, msCache := cacheTxContext(ctx, qMsg.TxId, qMsg.MsgId, qMsg.BlockHeight)
```

Comment:
How do you compare calling staking keeper functions directly v.s. using the msgSvr functions in the `runUnwrappedMsg` function?

The function for constructinig the cache tx makes me worried a bit as I'm not sure whether txid/msgid/blockheight are the only things we need there

Reply:
My bad, just realised that this function was same as the previous implementation and it's from baseapp... All good